src/constructs/ecs/ecs-task.ts

```diff
@@ -139,112 +138,120 @@ export class GuEcsTask {
   stateMachine: StateMachine;
 
   constructor(scope: GuStack, id: string, props: GuEcsTaskProps) {
-    const timeout = props.taskTimeoutInMinutes ?? 15;
+    const {
+      app,
 
-    // see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecs-taskdefinition.html#cfn-ecs-taskdefinition-cpu for details
-    const defaultCpu = 2048; // 2 cores and from 4-16GB memory
-    const defaultMemory = 4096; // 4GB
+      // see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecs-taskdefinition.html#cfn-ecs-taskdefinition-cpu for details
+      cpu = 2048, // 2 cores and from 4-16GB memory
```

Comment:
Somehow I never realised defaults with destructuring was possible! But it's really nice.

Reply:
Yeah it's really nice! I was inspired by an [old HTTP203 episode](https://youtu.be/te3Zm4bHBVs?t=240).